api/server/router/container/exec.go

```diff
@@ -114,39 +114,40 @@ func (s *containerRouter) postContainerExecStart(ctx context.Context, w http.Res
 			if !execStartCheck.Tty && versions.GreaterThanOrEqualTo(httputils.VersionFromContext(ctx), "1.42") {
 				contentType = types.MediaTypeMultiplexedStream
 			}
-			fmt.Fprint(outStream, "HTTP/1.1 101 UPGRADED\r\nContent-Type: "+contentType+"\r\nConnection: Upgrade\r\nUpgrade: tcp\r\n")
+			_, _ = fmt.Fprint(outStream, "HTTP/1.1 101 UPGRADED\r\nContent-Type: "+contentType+"\r\nConnection: Upgrade\r\nUpgrade: tcp\r\n")
 		} else {
-			fmt.Fprint(outStream, "HTTP/1.1 200 OK\r\nContent-Type: application/vnd.docker.raw-stream\r\n")
+			_, _ = fmt.Fprint(outStream, "HTTP/1.1 200 OK\r\nContent-Type: application/vnd.docker.raw-stream\r\n")
 		}
 
 		// copy headers that were removed as part of hijack
 		if err := w.Header().WriteSubset(outStream, nil); err != nil {
 			return err
 		}
-		fmt.Fprint(outStream, "\r\n")
+		_, _ = fmt.Fprint(outStream, "\r\n")
 
-		stdin = inStream
-		stdout = outStream
-		if !execStartCheck.Tty {
+		if execStartCheck.Tty {
+			stdin = inStream
+			stdout = outStream
+		} else {
 			stderr = stdcopy.NewStdWriter(outStream, stdcopy.Stderr)
 			stdout = stdcopy.NewStdWriter(outStream, stdcopy.Stdout)
 		}
 	}
 
-	options := container.ExecStartOptions{
+	// Now run the user process in container.
+	//
+	// TODO: Maybe we should we pass ctx here if we're not detaching?
+	err := s.backend.ContainerExecStart(context.Background(), execName, container.ExecStartOptions{
 		Stdin:       stdin,
 		Stdout:      stdout,
 		Stderr:      stderr,
 		ConsoleSize: execStartCheck.ConsoleSize,
-	}
-
-	// Now run the user process in container.
-	// Maybe we should we pass ctx here if we're not detaching?
-	if err := s.backend.ContainerExecStart(context.Background(), execName, options); err != nil {
+	})
+	if err != nil {
 		if execStartCheck.Detach {
 			return err
 		}
-		stdout.Write([]byte(err.Error() + "\r\n"))
+		_, _ = fmt.Fprintf(stdout, "%v\r\n", err)
```

Comment:
LOL, my IDE had a bit of a false positive here; it kept complaining "err can be nil!" (not taking the `if err != nil` check above into account. 

As all other code was using `fmt.Fprintf`, I decided to just go with that to also make it more consistent.

Reply:
Hm... wondering if this relates to the failure;

```
=== Failed
=== FAIL: amd64.integration-cli TestDockerCLIExecSuite/TestExecInteractive (0.34s)
    docker_cli_exec_test.go:64: assertion failed: error is not nil: EOF
    --- FAIL: TestDockerCLIExecSuite/TestExecInteractive (0.34s)
```